test/Microsoft.AspNetCore.Mvc.Razor.Test/ViewComponentTagHelpers/ViewComponentTagHelperDescriptorFactoryTest.cs

```diff
@@ -0,0 +1,194 @@
+                Assert.Equal(descriptor.TagStructure, expectedDescriptor.TagStructure);
+
+                // Check attributes.
+                Assert.Equal(descriptor.Attributes.Count(), expectedDescriptor.Attributes.Count());
+                for (var i = 0; i < descriptor.Attributes.Count(); i++)
+                {
+                    var attribute = descriptor.Attributes.ElementAt(i);
+                    var expectedAttribute = expectedDescriptor.Attributes.ElementAt(i);
+
+                    Assert.Equal(attribute.Name, expectedAttribute.Name);
+                    Assert.Equal(attribute.PropertyName, expectedAttribute.PropertyName);
+                    Assert.Equal(attribute.TypeName, expectedAttribute.TypeName);
+                }
+  
+                // Check required attributes.
+                Assert.Equal(descriptor.RequiredAttributes.Count(), expectedDescriptor.RequiredAttributes.Count());
+                for (int i = 0; i < descriptor.RequiredAttributes.Count(); i++)
+                {
+                    var requiredAttribute = descriptor.RequiredAttributes.ElementAt(i);
+                    var expectedRequiredAttribute = expectedDescriptor.RequiredAttributes.ElementAt(i);
+
+                    Assert.Equal(requiredAttribute.Name, expectedRequiredAttribute.Name);
+                }
+            }
+        }
+
+        [Fact]
+        public void CreateDescriptors_ThrowsException_OnNullAssemblyName()
+        {
+            // Arrange
+            var provider = new TestViewComponentDescriptorProvider();
+            var factory = new ViewComponentTagHelperDescriptorFactory(provider);
+
+            // Act, Assert
+            Assert.Throws<ArgumentNullException>(() => factory.CreateDescriptors(null));
+        }
+
+        private class TestViewComponentDescriptorProvider : IViewComponentDescriptorProvider
+        {
+            private ViewComponentDescriptor _viewComponentDescriptorOne = new ViewComponentDescriptor
+            {
+                DisplayName = "OneDisplayName",
+                FullName = "OneViewComponent",
+                ShortName = "One",
+                MethodInfo = typeof(ViewComponentTagHelperDescriptorFactoryTest)
+                    .GetMethod("TestInvokeOne"),
+                TypeInfo = typeof(ViewComponentTagHelperDescriptorFactory).GetTypeInfo()
+            };
+
+            private ViewComponentDescriptor _viewComponentDescriptorTwo = new ViewComponentDescriptor
+            {
+                DisplayName = "TwoDisplayName",
+                FullName = "TwoViewComponent",
+                ShortName = "Two",
+                MethodInfo = typeof(ViewComponentTagHelperDescriptorFactoryTest)
+                    .GetMethod("TestInvokeTwo"),
+                TypeInfo = typeof(ViewComponentTagHelperDescriptorFactoryTest).GetTypeInfo()
+            };
+
+            public TagHelperDescriptor tagHelperDescriptorOne = new TagHelperDescriptor
```

Comment:
http://stackoverflow.com/questions/150479/order-of-items-in-classes-fields-properties-constructors-methods